lib/libc/picolibc/Kconfig

```diff
@@ -5,7 +5,7 @@ if PICOLIBC
 
 config PICOLIBC_USE_MODULE
 	bool "Picolibc as module"
-	default y if "$(TOOLCHAIN_HAS_PICOLIBC)" != "y"
+	default y if ("$(TOOLCHAIN_HAS_PICOLIBC)" != "y") && TOOLCHAIN_CAN_BUILD_PICOLIBC_MODULE
```

Comment:
this is only a default, meaning it can be changed.

But if `PICOLIBC==y` and the toolchain can only build picolibc as module, but has `TOOLCHAIN_HAS_PICOLIBC==n`, then this setting must be a forced yes.
Either by being promptless, or with a `select PICOLIBC_USE_MODULE` somewhere.

Reply:
I use `TOOLCHAIN_SUPPORTS_PICOLIBC_MODULE` to select `PICOLIBC_USE_MODULE` instead.